### PR TITLE
Adding Carbon Monoxide Detector input

### PR DESCRIPTION
**Pull request by wjsroot**
I know you do this voluntarily. I'm trying not to make more work for you. I'd like to have a GPIO input tied to a carbon monoxide detector. I've forked your code, made the changes that I believe are needed (based on other devices which register as CO detectors) but I've failed at getting the code into a state where I can test it. My Node and Git skills are practically nonexistent. If possible, please disregard the second commit.
Please let me know if there's anything I should be doing differently or anything I can do to help. Thanks for making it this far.

### CODE REVIEW

File: config.schema.json [hunk 1]
```diff
@@ -154,6 +154,12 @@
                           "smoke"
                         ]
                       },
+                      {
+                        "title": "Carbon Monoxide Sensor",
+                        "enum": [
+                          "carbon"
```
Comment (ebaauw) [inline config.schema.json:160]: Please use `co` or `carbonmonoxide` your distinguish from potential future CO2 sensors.

Comment (ebaauw) [inline config.schema.json:160]: Please keep the alphabetical sort order by type.  Also in the other lists and descriptions.

File: lib/RpiAccessory/index.js [hunk 2]
```diff
@@ -341,7 +341,14 @@ class RpiAccessory extends homebridgeLib.AccessoryDelegate {
       gpioAccessory.emit('initialised')
     })
   }
-
+  async addCarbon (device) {
```
Comment (ebaauw) [inline lib/RpiAccessory/index.js:344]: Please call this `addCarbonMonoxide()` to leave room for future CO2 sensors.

File: lib/RpiPlatform.js [hunk 3]
```diff
@@ -86,6 +86,7 @@ class RpiPlatform extends homebridgeLib.Platform {
             case 'leak':
             case 'motion':
             case 'smoke':
+            case 'carbon':
```
Comment (ebaauw) [inline lib/RpiPlatform.js:89]: Please keep sort order.

File: lib/RpiService/GpioInput/GpioCarbon.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// homebridge-rpi/lib/RpiService/GpioInput/GpioCarbon.js
+// Copyright © 2019-2022 Erik Baauw and Bill Stoddart.  All rights reserved.
+//
+// Homebridge plugin for Raspberry Pi.
+
+'use strict'
+
+const GpioInput = require('./index.js')
+
+class GpioCarbon extends GpioInput {
```
Comment (ebaauw) [inline lib/RpiService/GpioInput/GpioCarbon.js:10]: Please call this `GpioCarbonMonoxide`.

File: lib/RpiService/GpioInput/index.js [hunk 5]
```diff
@@ -19,6 +19,7 @@ class GpioInput extends homebridgeLib.ServiceDelegate {
   static get GpioMotion () { return require('./GpioMotion') }
   static get GpioRocker () { return require('./GpioRocker') }
   static get GpioSmoke () { return require('./GpioSmoke') }
+  static get GpioCarbon () { return require('./GpioCarbon') }
```
Comment (ebaauw) [inline lib/RpiService/GpioInput/index.js:22]: Please maintain the sort order.

File: lib/RpiService/index.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ const homebridgeLib = require('homebridge-lib')
 class RpiService extends homebridgeLib.ServiceDelegate {
   static get PowerLed () { return require('./RpiPowerLed') }
   static get SmokeSensor () { return require('./RpiSmokeSensor') }
+  static get CarbonSensor () { return require('./RpiCarbonSensor') }
```
Comment (ebaauw) [inline lib/RpiService/index.js:13]: You don’t need this - these are services in the accessory for the Raspberry Pi.

File: lib/RpiService/index.js [hunk 7]
```diff
@@ -23,6 +24,7 @@ class RpiService extends homebridgeLib.ServiceDelegate {
   static get GpioRocker () { return require('./GpioInput').GpioRocker }
   static get GpioServo () { return require('./GpioServo') }
   static get GpioSmoke () { return require('./GpioInput').GpioSmoke }
+  static get GpioCarbon () { return require('./GpioInput').GpioCarbon }
```
Comment (ebaauw) [inline lib/RpiService/index.js:27]: Please maintain the sort order.

File: package.json [hunk 8]
```diff
@@ -30,12 +30,12 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/ebaauw/homebridge-rpi.git"
+    "url": "git+https://github.com/wjsroot/homebridge-rpi.git"
```
Comment (ebaauw) [inline package.json:33]: You would need this only if you want to publish an `npm` package under your own name.  In that case, you would have to change the package name as well, e.g. to `@wjsroot/homebridge-rpi`, as the `npm` repository requires package names to be unique.

**Review by ebaauw — CHANGES_REQUESTED** [review]
Thanks for the PR.  Always nice to see others taking my software a step further!

For testing during development, I create a symlink in `/usr/lib/node_modules` (or `/usr/local/lib/node_modules`, depending on your NodeJS installation) to the `homebridge-rpi`directory.  Make sure to run `npm i` in that directory to install the dependencies, and you should be good to go.  Only for testing on a separate server, I release a beta version through `npm`.

I think you understand the structure of Homebridge RPi quite well, there’s a lot of places where a new sensor type needs to be added, but at least it’s pretty straightforward.

I’m bitching a bit about naming and keeping the code sorted, but overall it looks OK.

**Comment by wjsroot** [comment]
I greatly appreciate your help and feedback. Please let me know how this looks, I think I've gotten everything. I changed 'carbon' to 'carbonmonoxide'; this makes a lot of sense, I even have sensors that detect carbon dioxide, smart to add clarity.

**Review by ebaauw — APPROVED** [review]

**Comment by ebaauw** [comment]
In v1.6.4.

**Pull request merged**